<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Arunabha Majumdar, Tanushree Haldar" />


<title>MPGE package: A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      code.sourceCode > span { display: inline-block; line-height: 1.25; }
  code.sourceCode > span { color: inherit; text-decoration: inherit; }
  code.sourceCode > span:empty { height: 1.2em; }
  .sourceCode { overflow: visible; }
  code.sourceCode { white-space: pre; position: relative; }
  div.sourceCode { margin: 1em 0; }
  pre.sourceCode { margin: 0; }
  @media screen {
  div.sourceCode { overflow: auto; }
  }
  @media print {
  code.sourceCode { white-space: pre-wrap; }
  code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
  }
  pre.numberSource code
    { counter-reset: source-line 0; }
  pre.numberSource code > span
    { position: relative; left: -4em; counter-increment: source-line; }
  pre.numberSource code > span > a:first-child::before
    { content: counter(source-line);
      position: relative; left: -1em; text-align: right; vertical-align: baseline;
      border: none; display: inline-block;
      -webkit-touch-callout: none; -webkit-user-select: none;
      -khtml-user-select: none; -moz-user-select: none;
      -ms-user-select: none; user-select: none;
      padding: 0 4px; width: 4em;
      color: #aaaaaa;
    }
  pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
  div.sourceCode
    {   }
  @media screen {
  code.sourceCode > span > a:first-child::before { text-decoration: underline; }
  }
  code span.al { color: #ff0000; font-weight: bold; } /* Alert */
  code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
  code span.at { color: #7d9029; } /* Attribute */
  code span.bn { color: #40a070; } /* BaseN */
  code span.bu { } /* BuiltIn */
  code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
  code span.ch { color: #4070a0; } /* Char */
  code span.cn { color: #880000; } /* Constant */
  code span.co { color: #60a0b0; font-style: italic; } /* Comment */
  code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
  code span.do { color: #ba2121; font-style: italic; } /* Documentation */
  code span.dt { color: #902000; } /* DataType */
  code span.dv { color: #40a070; } /* DecVal */
  code span.er { color: #ff0000; font-weight: bold; } /* Error */
  code span.ex { } /* Extension */
  code span.fl { color: #40a070; } /* Float */
  code span.fu { color: #06287e; } /* Function */
  code span.im { } /* Import */
  code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
  code span.kw { color: #007020; font-weight: bold; } /* Keyword */
  code span.op { color: #666666; } /* Operator */
  code span.ot { color: #007020; } /* Other */
  code span.pp { color: #bc7a00; } /* Preprocessor */
  code span.sc { color: #4070a0; } /* SpecialChar */
  code span.ss { color: #bb6688; } /* SpecialString */
  code span.st { color: #4070a0; } /* String */
  code span.va { color: #19177c; } /* Variable */
  code span.vs { color: #4070a0; } /* VerbatimString */
  code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">MPGE package: A two-step approach to testing overall effect of gene-environment interaction for multiple phenotypes</h1>
<h4 class="author">Arunabha Majumdar, Tanushree Haldar</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Interaction between a genetic variant (e.g., a SNP) and an environmental variable (e.g., physical activity) can have a shared effect on multiple phenotypes (e.g., LDL and HDL). MPGE is a two-step method to test for an overall interaction effect on multiple phenotypes. In first step, the method tests for an overall marginal genetic association between the genetic variant and the multivariate phenotype. In the second step, SNPs which show an evidence of marginal overall genetic effect in the first step are prioritized while testing for an overall GxE effect. That is, a more liberal threshold of significance level is considered in the second step while testing for an overall GxE effect for these promising SNPs compared to the other SNPs.</p>
<p>This R-package consists of the following main functions:</p>
<ol style="list-style-type: decimal">
<li>mv_G_GE: for a batch of genetic variants, this function provides two different p-values for each genetic variant, one from the test of marginal overall genetic association with multiple phenotypes, and the other from the test of overall GxE effect on multivariate phenotype allowing for a possible marginal effect due to the genetic variant and a marginal effect due to the environmental variable.</li>
<li>WHT: this function implements the weighted multiple hypothesis testing procedure to adjust for multiple testing while combining the two steps of testing gene-environment interaction in the two-step GxE testing procedure, given two sets of p-values obtained using the previous function mv_G_GE for genome-wide genetic variants.</li>
<li>SST: this function implements the subset multiple hypothesis testing procedure to adjust for multiple testing while combining the two steps of testing gene-environment interaction based on the same two sets of p-values described above.</li>
</ol>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>You can install MPGE from CRAN.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">#install.packages(&quot;MPGE&quot;)</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co">#library(&quot;MPGE&quot;)</span></span></code></pre></div>
</div>
<div id="run-mv_g_ge" class="section level1">
<h1>Run mv_G_GE</h1>
<p>We will demonstrate how to run the mv_G_GE function. First, load the example data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(<span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co"># Load the phenotype data</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>phenofile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;phenotype_data.rda&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="kw">head</span>(phenotype_data)</span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co">##        pheno1     pheno2      pheno3</span></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="co">## 1  0.31921537  1.0531580  1.21120849</span></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="co">## 2 -0.67344929  0.9330864 -1.40170032</span></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co">## 3  0.79089818 -0.7405425  2.26619916</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="co">## 4  0.63810298  1.0210492  1.93624445</span></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="co">## 5 -0.02325848 -0.3910989  2.25363813</span></span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="co">## 6 -0.99669088 -1.8149841 -0.04107952</span></span></code></pre></div>
<p>Here phenotype_data is a data.frame with three columns for three phenotypes and the number of rows to be the number of individuals in the sample (500 in this toy data). Data for each phenotype provided must be adjusted individually for relevant covariates (e.g., age, sex, genetic ancestry) beforehand, and should follow a normal distribution.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">library</span>(<span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="co"># Load the genotype data</span></span>
<span id="cb3-3"><a href="#cb3-3"></a>genofile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;genotype_data.rda&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="kw">head</span>(genotype_data)</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co">##   rs1 rs2</span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="co">## 1   1   0</span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="co">## 2   1   0</span></span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="co">## 3   0   0</span></span>
<span id="cb3-9"><a href="#cb3-9"></a><span class="co">## 4   0   0</span></span>
<span id="cb3-10"><a href="#cb3-10"></a><span class="co">## 5   0   0</span></span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="co">## 6   1   0</span></span></code></pre></div>
<p>Here, genotype_data is a data.frame with the columns as SNPs (e.g., rs1 and rs2 here). The rows correspond to the 500 individuals in the same order as in the phenotype data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="kw">library</span>(<span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># Load the data for environmental variable</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>genofile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;environment_data.rda&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="kw">head</span>(environment_data)</span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="co">##   env</span></span>
<span id="cb4-6"><a href="#cb4-6"></a><span class="co">## 1   0</span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="co">## 2   0</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="co">## 3   0</span></span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="co">## 4   0</span></span>
<span id="cb4-10"><a href="#cb4-10"></a><span class="co">## 5   1</span></span>
<span id="cb4-11"><a href="#cb4-11"></a><span class="co">## 6   0</span></span>
<span id="cb4-12"><a href="#cb4-12"></a><span class="co"># For example, non-smoker coded as 0 and smoker coded as 1.</span></span></code></pre></div>
<p>Here, environment_data is a data frame with single column for the environmental variable. The order of the 500 individuals in the row must be the same as provided in the phenotype and genotype data. Here, the environmental variable has two categories which were coded as 1 and 0 (e.g., smokers and non-smokers). Instead of numeric values, these can also be considered to be factors in the absence of a defined order in the categories. Now, given the required phenotype, genotype and environmental data, we run the mv_G_GE function next.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co">#Compute the p-value of testing marginal multivariate genetic association. And, compute the p-value of testing multivariate GxE effect in presence of possible marginal genetic effect and marginal environmental effect on the phenotypes.</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>result &lt;-<span class="st"> </span><span class="kw">mv_G_GE</span>(phenotype_data, genotype_data, environment_data)</span>
<span id="cb5-3"><a href="#cb5-3"></a>result</span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="co">##   SNP         G.P       GE.P</span></span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="co">## 1 rs1 0.000151109 0.00082551</span></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co">## 2 rs2 0.176325958 0.11378000</span></span></code></pre></div>
<p>The output (‘result’) of mv_G_GE is a data.frame. Each row of the output provides the pair of p-values for each genetic variant, first one (G.P) from the test of marginal overall genetic association, and the second one (GE.P) from the test of overall GxE effect in presence of possible marginal genetic effect and marginal environmental effect.</p>
</div>
<div id="run-wht-and-sst" class="section level1">
<h1>Run WHT and SST</h1>
<p>Next, we demonstrate how to run WHT and SST. First, load the example data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="kw">library</span>(<span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="co"># Load the p-values</span></span>
<span id="cb6-3"><a href="#cb6-3"></a>pvalues &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;mv_G_GxE_pvalues.rda&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;MPGE&quot;</span>)</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="kw">head</span>(mv_G_GxE_pvalues)</span>
<span id="cb6-5"><a href="#cb6-5"></a><span class="co">##   SNP          G.P       GE.P</span></span>
<span id="cb6-6"><a href="#cb6-6"></a><span class="co">## 1 rs1 1.023551e-27 1.3292e-02</span></span>
<span id="cb6-7"><a href="#cb6-7"></a><span class="co">## 2 rs2 1.540443e-23 3.2022e-05</span></span>
<span id="cb6-8"><a href="#cb6-8"></a><span class="co">## 3 rs3 2.726237e-08 3.3349e-04</span></span>
<span id="cb6-9"><a href="#cb6-9"></a><span class="co">## 4 rs4 7.730439e-30 2.6798e-03</span></span>
<span id="cb6-10"><a href="#cb6-10"></a><span class="co">## 5 rs5 2.173659e-22 1.0266e-05</span></span>
<span id="cb6-11"><a href="#cb6-11"></a><span class="co">## 6 rs6 1.476325e-14 4.8417e-06</span></span></code></pre></div>
<p>Here, mv_G_GxE_pvalues is a data.frame with three columns. First column lists the set of 1000 genetic variants. Second column provides the vector of p-values obtained from testing the marginal multivariate genetic association for these SNPs. And the third column provides the vector of p-values obtained from testing the overall GxE effect in presence of possible marginal genetic effect and marginal environmental effect. Thus, the input data has the same structure as the output produced by the first function mv_G_GE. Next, we run WHT for this example data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#Run WHT to implement the weighted hypothesis testing to adjust for multiple comparison, and find the significant SNPs with an overall GxE effect.</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>result &lt;-<span class="st"> </span><span class="kw">WHT</span>(mv_G_GxE_pvalues)</span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="kw">str</span>(result)</span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">## List of 2</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">##  $ GEsnps     : chr [1:10] &quot;rs2&quot; &quot;rs5&quot; &quot;rs6&quot; &quot;rs8&quot; ...</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">##  $ adjusted.PV:&#39;data.frame&#39;: 1000 obs. of  2 variables:</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">##   ..$ SNP  : chr [1:1000] &quot;rs1&quot; &quot;rs2&quot; &quot;rs3&quot; &quot;rs4&quot; ...</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">##   ..$ adj.P: num [1:1000] 2.12672 0.02049 0.85373 0.42877 0.00657 ...</span></span></code></pre></div>
<p>The output is a list, first element of which (‘GEsnps’) is a character vector providing the set of SNPs found to have a significant signal of overall GxE effect. Second element is a dataframe providing the adjusted GxE test p-values (with the corresponding genetic variants in the first column) obtained by the weighted hypothesis testing procedure adjusting for multiple comparison.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co">#Run SST to implement the subset testing to adjust for multiple comparison, and find the significant SNPs with an overall GxE effect.</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>result &lt;-<span class="st"> </span><span class="kw">SST</span>(mv_G_GxE_pvalues)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw">str</span>(result)</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="co">##  chr [1:11] &quot;rs2&quot; &quot;rs3&quot; &quot;rs5&quot; &quot;rs6&quot; &quot;rs8&quot; &quot;rs9&quot; &quot;rs10&quot; &quot;rs11&quot; &quot;rs12&quot; &quot;rs13&quot; ...</span></span></code></pre></div>
<p>Here, we run SST for the same example data analyzed by WHT function above. The output from SST is a character vector providing the set of SNPs having a significant signal of overall GxE effect identified by subset testing to adjust for the multiple comparison.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
